pkg/engine/variables/vars.go

```diff
@@ -94,6 +94,16 @@ func subValR(log logr.Logger, ctx context.EvalInterface, valuePattern string, pa
 			variable := strings.ReplaceAll(v, "{{", "")
 			variable = strings.ReplaceAll(variable, "}}", "")
 			variable = strings.TrimSpace(variable)
+
+			var substitutedVar interface{}
+			if operation, err := ctx.Query("request.operation"); err != nil {
+				return nil, fmt.Errorf("failed to resolve %v at path %s", variable, path)
+			} else {
+				if operation == "DELETE" {
+					variable = strings.Replace(variable, "request.object", "request.oldObject", 1)
```

Comment:
Should we replace all `request.object`? It could have nested variables defined.

Reply:
It replaces only first occurrence so it should work fine.
Could you please provide a case when it fails?

Reply:
@realshuting, I changed it to ReplaceAll. Please check